src/paulie/application/pauli_compiler_optimal.py

```diff
@@ -0,0 +1,612 @@
+﻿"""
+Optimal PauliCompiler
+"""
+# noqa
+from __future__ import annotations
+
+from dataclasses import dataclass
+from itertools import permutations
+from collections import deque
+from typing import Iterable
+
+from paulie.common.pauli_string_bitarray import PauliString
+from paulie.common.pauli_string_factory import get_identity, get_pauli_string
+
+
+def _single(n: int, i: int, label: str) -> PauliString:
```

Comment:
Should these "helper" methods be put into a more general file instead of directly within this file?